tests/unit/submitter/test_biosamples_submitter_service.py

```diff
@@ -0,0 +1,32 @@
+import json
+import unittest
+from os.path import dirname
+from unittest.mock import MagicMock
+
+from submission_broker.submission.entity import Entity
+
+from submitter.biosamples_submitter_service import BioSamplesSubmitterService
+
+
+class BioSamplesSubmitterServiceTest(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.archive_client = MagicMock()
+        self.submitter_service = BioSamplesSubmitterService(self.archive_client)
+
+        with open(dirname(__file__) + '/../../resources/biomaterial_entity.json') as file:
+            self.biomaterial: Entity = Entity('biomaterials', 0, attributes=json.load(file))
+
+    def test_update_sample_with_biostudies_accession(self):
+        biostudies_accession = 'BST-123'
+
+        self.submitter_service.update_sample_with_biostudies_accession(self.biomaterial, biostudies_accession)
+
+        actual_external_references = self.biomaterial.attributes.get('externalReferences')
+
+        self.assertEqual(len(actual_external_references), 1)
```

Comment:
Have you considered using [assertpy](https://github.com/assertpy/assertpy) for writing complex assertions?
This makes the code as well as the error message cleaner.

Reply:
This assertion is not complex and it looks clean enough to me.

Reply:
I am going to try to use it from now